ios/engine/KMEI/KeymanEngine/Classes/Manager.swift

```diff
@@ -963,4 +963,64 @@ public class Manager: NSObject, UIGestureRecognizerDelegate {
     shouldReloadKeyboard = true
     inputViewController.reload()
   }
+
+  /*-----------------------------
+   *    Legacy API endpoints    -
+   *-----------------------------
+   *
+   * Some functionality has been refactored into separate classes for 12.0.
+   * No reason we can't add a couple of helper functions to help forward
+   * the data for apps built on older versions of KMEI, though.
+   */
+
+  public func downloadKeyboard(from url: URL) {
+    ResourceDownloadManager.shared.downloadKeyboard(from: url)
+  }
+
+  public func downloadKeyboard(withID: String, languageID: String, isUpdate: Bool, fetchRepositoryIfNeeded: Bool = true) {
+    ResourceDownloadManager.shared.downloadKeyboard(withID: withID,
+                                                    languageID: languageID,
+                                                    isUpdate: isUpdate,
+                                                    fetchRepositoryIfNeeded: fetchRepositoryIfNeeded)
+  }
+
+  // A new API, but it so closely parallels downloadKeyboard that we should add a 'helper' handler here.
+  public func downloadLexicalModel(from url: URL) {
+    ResourceDownloadManager.shared.downloadLexicalModel(from: url)
+  }
+
+  // A new API, but it so closely parallels downloadKeyboard that we should add a 'helper' handler here.
+  public func downloadLexicalModel(withID: String, languageID: String, isUpdate: Bool, fetchRepositoryIfNeeded: Bool = true) {
+    ResourceDownloadManager.shared.downloadLexicalModel(withID: withID,
+                                                        languageID: languageID,
+                                                        isUpdate: isUpdate,
+                                                        fetchRepositoryIfNeeded: fetchRepositoryIfNeeded)
+  }
+
+  public func stateForKeyboard(withID keyboardID: String) -> KeyboardState {
+    return ResourceDownloadManager.shared.stateForKeyboard(withID: keyboardID)
+  }
+
+  // Technically new, but it does closely parallel an old API point.
+  public func stateForLexicalModel(withID modelID: String) -> KeyboardState {
+    return ResourceDownloadManager.shared.stateForLexicalModel(withID: modelID)
+  }
+
+  public func parseKMP(_ folder: URL, type: LanguageResourceType = .keyboard) throws -> Void {
+    switch type {
+      case .keyboard:
+        try! parseKbdKMP(folder)
+        break
+      case .lexicalModel:
+        // Yep.  Unlike the original, THIS one is static.
+        try! Manager.parseLMKMP(folder)
+        break
+    }
+  }
+
+  // To re-implement the old API (missing from v11!) we need to pick one version of height.
+  // For library users, the total height of the InputViewController should be most useful.
+  public func keyboardHeight() -> CGFloat {
```

Comment:
For maintainability, can we have more markings to clarify where legacy API calls start and stop?
Should "new" API's just be treated as API's?

I can't tell if `stateForKeyboard`, `stateForLexicalModel` and `parseKMP` are part of the API

Reply:
I'm pretty sure I marked each one that is "new" - isn't that what you're asking for?

For the ones you listed:

- [stateForKeyboard](https://help.keyman.com/DEVELOPER/engine/iphone-and-ipad/11.0/Manager/stateForKeyboard.php)
- `stateForLexicalModel` is new; lexical models weren't around pre v12
- [parseKMP](https://help.keyman.com/DEVELOPER/engine/iphone-and-ipad/11.0/Manager/parseKMP.php)

Reply:
I'm saying an iOS developer may not always know to check what's documented on help.keyman.com before moving things around :)

Reply:
... that's why this PR exists.  Things _**were**_ moved around; this "un-moves" them.  Well, actually it just makes helpers in the old places that forward to the new.